src/support/threads.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2016 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+
+#include <iostream>
+
+#include "threads.h"
+
+
+// debugging tools
+
+#ifdef BINARYEN_THREAD_DEBUG
+std::mutex debug;
+#define DEBUG_PRINT(x) { std::lock_guard<std::mutex> lock(debug); x }
+#else
+#define DEBUG_PRINT(x)
+#endif
+
+
+namespace wasm {
+
+// Global thread information
+
+bool setMainThreadId = false;
```

Comment:
hmm, i wasn't familiar with that so I read up on it now. I'm not sure how to use it here - we need it to start its life initialized to false, before any code runs (it might be read by a global ctor), and that class doesn't allow ` = false`?